pkg/router/template/router.go

```diff
@@ -222,18 +223,61 @@ func (r *templateRouter) EnableRateLimiter(interval int, handlerFunc ratelimiter
 	glog.V(2).Infof("Template router will coalesce reloads within %v seconds of each other", interval)
 }
 
+// When there is no default cert, cobble one together from
+// the mounted secret, if present
+func secretToPem(outName string) error {
+	// The secret, when present, is mounted on /etc/pki/tls/private
+	// The secret has two components crt.tls and key.tls
+	// When the default cert is provided by the admin it is a pem
+	//   tls.crt is the supplied pem and tls.key is the key
+	//   extracted from the pem
+	// When the admin does not provide a default cert, the secret
+	//   is created via the service annotation. In this case
+	//   tls.crt is the cert and tls.key is the key
+	//   The crt and key are concatenated to form the needed pem
+	// is /etc/pki/tls/private/tls.crt and tls.key present
+	var fileCrtName = "/etc/pki/tls/private/tls.crt"
+	var fileKeyName = "/etc/pki/tls/private/tls.key"
+	pemBlock, err := ioutil.ReadFile(fileCrtName)
+	if err != nil {
+		return err
+	}
+	keys, err := cmdutil.PrivateKeysFromPEM(pemBlock)
+	if err != nil {
+		return err
+	}
+	if len(keys) == 0 {
+		// Try to get the key from the tls.key file
+		keyBlock, err := ioutil.ReadFile(fileKeyName)
+		if err != nil {
+			return err
+		}
+		pemBlock = append(pemBlock, keyBlock...)
+	}
+	return ioutil.WriteFile(outName, pemBlock, 0444)
+}
+
 // writeDefaultCert is called a single time during init to write out the default certificate
 func (r *templateRouter) writeDefaultCert() error {
+	dir := filepath.Join(r.dir, certDir)
+	outPath := filepath.Join(dir, fmt.Sprintf("%s.pem", defaultCertName))
 	if len(r.defaultCertificate) == 0 {
+		err := secretToPem(outPath)
```

Comment:
nevermind, @knobunc is right -- we need a case for if `r.defaultCertificatePath` is set inside the case where `r.defaultCertificate` is empty so we don't error out in a valid case.